worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -1776,6 +1776,79 @@ public int makeCylinder(BlockVector3 pos, Pattern block, double radiusX, double
         return affected;
     }
 
+    /**
+     * Makes a cone.
+     *
+     * @param pos Center of the cone
+     * @param block The block pattern to use
+     * @param radiusX The cone's largest north/south extent
+     * @param radiusZ The cone's largest east/west extent
+     * @param height The cone's up/down extent. If negative, extend downward.
+     * @param filled If false, only a shell will be generated.
+     * @param thickness The cone's wall thickness, if it's hollow.
+     * @return number of blocks changed
+     * @throws MaxChangedBlocksException thrown if too many blocks are changed
+     */
+    public int makeCone(BlockVector3 pos, Pattern block, double radiusX, double radiusZ, int height, boolean filled,
+                        double thickness) throws MaxChangedBlocksException {
+        int affected = 0;
+
+        final int ceilRadiusX = (int) Math.ceil(radiusX);
+        final int ceilRadiusZ = (int) Math.ceil(radiusZ);
+
+        for (int y = 0; y < height; ++y) {
+            forX:
+            for (int x = 0; x <= ceilRadiusX; ++x) {
+
+                forZ:
+                for (int z = 0; z <= ceilRadiusZ; ++z) {
+
+                    double xSquaredOverRadiusX = Math.pow(x, 2) /  Math.pow(radiusX, 2);
+                    double zSquaredOverRadiusZ = Math.pow(z, 2) / Math.pow(radiusZ, 2);
+                    double ySquaredMinusHeightOverHeightSquared = Math.pow(y - height, 2)
+                            / Math.pow(height, 2);
+                    double distanceFromOriginMinusHeightSquared = xSquaredOverRadiusX + zSquaredOverRadiusZ
+                            - ySquaredMinusHeightOverHeightSquared;
+
+                    if (distanceFromOriginMinusHeightSquared > 1) {
```

Comment:
You can technically even add a short circuit case outside of the z loop if the xSquared - ySquared is >1, as that guarantees this will be >1 too. Unsure how often that'll be the case though so the extra calculation might end up using more resources than it saves